torch/utils/data/dataloader.py

```diff
@@ -300,28 +300,28 @@ def __len__(self):
 class _BaseDataLoaderIter(object):
     def __init__(self, loader):
         self.dataset = loader.dataset
-        self.dataset_kind = loader.dataset_kind
-        self.auto_collation = loader._auto_collation
+        self._dataset_kind = loader._dataset_kind
+        self._auto_collation = loader._auto_collation
         self.drop_last = loader.drop_last
-        self.index_sampler = loader._index_sampler
+        self._index_sampler = loader._index_sampler
         self.num_workers = loader.num_workers
         self.pin_memory = loader.pin_memory and torch.cuda.is_available()
```

Comment:
FWIW changing `num_workers` or `pin_memory` might result in some pretty bad bugs too, so we might just as well make everything private 🤷‍♂

Reply:
I made everything private :)